wallet/sync.go

```diff
@@ -375,63 +375,38 @@ func (w *Wallet) scanAddressIndex(start int, end int, account uint32,
 		return uint32(lastUsed), addr, nil
 	}
 
-	// We can't find any used addresses. The wallet is
-	// unused.
-	return 0, nil, nil
-}
-
-// doAddressResync resyncs the address manager to a given address.
-func (w *Wallet) doAddressResync(addr dcrutil.Address, acct uint32, idx uint32,
-	internal bool) error {
-	isSynced := false
-	addrFunction := w.Manager.NextExternalAddresses
-	if internal {
-		addrFunction = w.Manager.NextInternalAddresses
-	}
-
-	counter := uint32(0)
-	for !isSynced {
-		// Generate some new addresses and scan them to see
-		// if any of the match the address to sync to.
-		addrs, err := addrFunction(acct, addrSeekWidth)
+	// In the case that 0 was returned as the last used address,
+	// make sure the the 0th address was not used. If it was,
+	// return this address to let the caller know that this
+	// 0th address was used.
+	if lastUsed == 0 {
+		addr, err := w.Manager.AddressDerivedFromDbAcct(0,
+			account, branch)
+		// Skip erroneous keys.
 		if err != nil {
-			return err
+			return 0, nil, err
 		}
 
-		for _, newAddr := range addrs {
-			if bytes.Compare(addr.ScriptAddress(),
-				newAddr.Address().ScriptAddress()) == 0 {
-				isSynced = true
-			}
+		exists, err := chainClient.ExistsAddress(addr)
+		if err != nil {
+			return 0, nil, fmt.Errorf("failed to access chain server: %v",
+				err.Error())
 		}
 
-		// Don't let this loop infinitely.
-		if counter > waddrmgr.MaxAddressesPerAccount/addrSeekWidth {
-			break
+		if exists {
+			return 0, addr, nil
 		}
-
-		log.Debugf("Currently getting address %v", counter*addrSeekWidth)
-
-		counter++
 	}
 
-	if isSynced {
-		return nil
-	}
-
-	return fmt.Errorf("failed to sync to address %v during address rescan",
-		addr.String())
+	// We can't find any used addresses. The wallet is
```

Comment:
Should be account and branch from args is unused?  doesn't mean entire wallet is unused